core/src/test/java/com/backblaze/b2/json/B2JsonTest.java

```diff
@@ -1801,4 +1837,77 @@ private SubclassH(String name) {
         }
     }
 
+    @Test
+    public void testRequiredFieldNotInVersion() throws B2JsonException {
+        final String json = "{}";
+        final B2JsonOptions options = B2JsonOptions.builder().setVersion(1).build();
+        final VersionedContainer obj = bzJson.fromJson(json, VersionedContainer.class, options);
+        assertEquals(0, obj.x);
+        assertEquals(1, obj.version);
+    }
+
+    @Test
+    public void testRequiredFieldMissingInVersion() throws B2JsonException {
+        final String json = "{}";
+        final B2JsonOptions options = B2JsonOptions.builder().setVersion(5).build();
+
+        thrown.expectMessage("required field x is missing");
```

Comment:
can we say which class has the problem?
please, pretend i'm asking this for all new error messages. :)

Reply:
This is not a new error message, and...

B2Json errors show through the public B2 APIs at Backblaze.  When a client passes in JSON with a missing field, this error message is returned.  The Java class name won't make sense to the caller.